cmd/broker/binding_test.go

```diff
@@ -116,3 +119,131 @@ func TestBinding(t *testing.T) {
 	})
 
 }
+
+func TestDeprovisioningWithExistingBindings(t *testing.T) {
+	// given
+	cfg := fixConfig()
+	// Disable EDP to have all steps successfully executed
+	cfg.EDP.Disabled = true
+	suite := NewBrokerSuiteTestWithConfig(t, cfg)
+	defer suite.TearDown()
+	iid := uuid.New().String()
+	bindingID1 := uuid.New().String()
+	bindingID2 := uuid.New().String()
+
+	response := suite.CallAPI("PUT", fmt.Sprintf("oauth/v2/service_instances/%s?accepts_incomplete=true", iid),
+		`{
+					"service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
+					"plan_id": "361c511f-f939-4621-b228-d0fb79a1fe15",
+					"context": {
+						"globalaccount_id": "g-account-id",
+						"subaccount_id": "sub-id",
+						"user_id": "john.smith@email.com"
+					},
+					"parameters": {
+						"name": "testing-cluster",
+						"region": "eu-central-1"
+					}
+		}`)
+	opID := suite.DecodeOperationID(response)
+	suite.processProvisioningByOperationID(opID)
+	suite.WaitForOperationState(opID, domain.Succeeded)
+
+	// when we create two bindings
+	response = suite.CallAPI("PUT", fmt.Sprintf("oauth/v2/service_instances/%s/service_bindings/%s", iid, bindingID1),
+		`{
+                "service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
+                "plan_id": "361c511f-f939-4621-b228-d0fb79a1fe15"
+               }`)
+	require.Equal(t, http.StatusCreated, response.StatusCode)
+
+	response = suite.CallAPI("PUT", fmt.Sprintf("oauth/v2/service_instances/%s/service_bindings/%s", iid, bindingID2),
+		`{
+                "service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
+                "plan_id": "361c511f-f939-4621-b228-d0fb79a1fe15"
+               }`)
+	require.Equal(t, http.StatusCreated, response.StatusCode)
+
+	// when we deprovision successfully
+	response = suite.CallAPI("DELETE", fmt.Sprintf("oauth/v2/service_instances/%s?accepts_incomplete=true&plan_id=361c511f-f939-4621-b228-d0fb79a1fe15&service_id=47c9dcbf-ff30-448e-ab36-d3bad66ba281", iid),
+		``)
+	deprovisioningID := suite.DecodeOperationID(response)
+	suite.FinishDeprovisioningOperationByProvisioner(deprovisioningID)
+	suite.WaitForInstanceRemoval(iid)
```

Comment:
Can we check after this instruction that GET binding works and returns previously created binding?

Reply:
In this test case we test if we can deprovision cluster with existing bindings. Testing if bindings work just after deprovisioning and instance removal fixes us on current behaviour (bindings removal by asynchronous process).

Reply:
Worked on separate issue.